Katib v1beta1 documentation update (#2312)
kubeflow/website: content/en/docs/components/katib/env-variables.md

+++
title = "Environment Variables for Katib Components"
description = "How to set up environment variables for each Katib component"
weight = 60
                    
+++

This guide describes environment variables for each Katib component.
If you want to change your Katib installation,
you can modify some of these variables.

In the tables below you can find descriptions, default values and mandatory
properties for all environment variables in each Katib component.
If a variable has a mandatory property, you need to set the relevant
environment variable in an appropriate Katib component's manifest.

## Katib Controller

> These are environment variables for the

Reviewer: ```suggestion
Below are the environment variables for the
```

                    
+++

> This page describes information about environment variables for each Katib component.

Reviewer: ```suggestion
This guide is about environment variables for each Katib component.
```

Author: SGTM, changed

> If you want to change Katib installation, you can modify some of these variables.

Reviewer: ```suggestion
If you want to change your Katib installation, you can modify some of these variables.
```

> In the tables below you can see description, default values and mandatory property for
> all environment variables in each Katib component. If variable has mandatory property,
> appropriate Katib component's manifest must have this environment variable.

Reviewer: Suggestions:

- Change to plural (because of "all variables" per component)
- Add indefinite articles ("a")

```suggestion
In the tables below there are descriptions, default values and mandatory properties for
all environment variables in each Katib component. If a variable has a mandatory property,
you need to set the relevant environment variable in an appropriate Katib component's manifest.
```

(Please check if the meaning is still the same - we're giving instructions to "you" (the reader), so changing to active voice)

(It may also make sense to link this to 80 char per line)

## Katib Controller

> This is the table of environment variables for the [Katib Controller](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/katib-controller.yaml) deployment.

Reviewer: ```suggestion
These are environment variables for the [Katib Controller](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/katib-controller/katib-controller.yaml) deployment:
```

## Katib UI

> This is the table of environment variables for the [Katib UI](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/ui/deployment.yaml) deployment.

Reviewer: ```suggestion
These are environment variables for the [Katib UI](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/ui/deployment.yaml) deployment:
```

## Katib UI

These are environment variables for the
[Katib UI](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/ui/deployment.yaml)

> deployment.

Reviewer: ```suggestion
deployment:
```

## Katib DB Manager

These are environment variables for the
[Katib DB Manager](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/db-manager/deployment.yaml)

> deployment.

Reviewer: ```suggestion
deployment:
```

Currently, Katib DB Manager supports only **MySQL** database. You can use your own DB Manager and Database to report metrics.

For the [Katib DB Manager](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/db-manager/deployment.yaml#L29)
you can change `DB_PASSWORD` to your own MySQL DB password.

Katib DB Manager creates connection to the DB, using `mysql` driver and this data source name:

**`DB_USER:DB_PASSWORD@tcp(KATIB_MYSQL_DB_HOST:KATIB_MYSQL_DB_PORT)/KATIB_MYSQL_DB_DATABASE?timeout=5s`**

## Katib MySQL DB

For the [Katib MySQL](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/mysql-db/deployment.yaml)

> we set these environment variables:

Reviewer: ```suggestion
you need to set these environment variables:
```

Using the second person ("you" - the user) is preferred

- `MYSQL_ROOT_PASSWORD` to a value from [katib-mysql-secrets](https://github.com/kubeflow/katib/blob/master/manifests/v1beta1/mysql-db/secret.yaml),
  which is equal to "test".
- `MYSQL_ALLOW_EMPTY_PASSWORD` as `true`
- `MYSQL_DATABASE` as `katib`.

> Check [here](https://github.com/docker-library/docs/tree/master/mysql#environment-variables)
> about all environment variables for the MySQL Docker image.

Reviewer: ```suggestion
You can refer to the list of 
[all environment variables](https://github.com/docker-library/docs/tree/master/mysql#environment-variables)
for the MySQL Docker image.
```
